plugins/chat/lib/chat/duplicate_message_validator.rb

```diff
@@ -2,45 +2,26 @@
 
 module Chat
   class DuplicateMessageValidator
-    attr_reader :chat_message
+    attr_reader :chat_message, :chat_channel
 
     def initialize(chat_message)
       @chat_message = chat_message
+      @chat_channel = chat_message&.chat_channel
     end
 
     def validate
-      return if SiteSetting.chat_duplicate_message_sensitivity.zero?
-      matrix =
-        DuplicateMessageValidator.sensitivity_matrix(SiteSetting.chat_duplicate_message_sensitivity)
+      return if chat_message.nil? || chat_channel.nil?
+      return if chat_message.user.bot?
+      return if chat_channel.direct_message_channel? && chat_channel.user_count <= 2
 
-      # Check if the length of the message is too short to check for a duplicate message
-      return if chat_message.message.length < matrix[:min_message_length]
-
-      # Check if there are enough users in the channel to check for a duplicate message
-      return if (chat_message.chat_channel.user_count || 0) < matrix[:min_user_count]
-
-      # Check if the same duplicate message has been posted in the last N seconds by any user
-      if !chat_message
-           .chat_channel
+      if chat_channel
            .chat_messages
-           .where("created_at > ?", matrix[:min_past_seconds].seconds.ago)
-           .where(message: chat_message.message)
+           .where(created_at: 30.seconds.ago..)
```

Comment:
1. I'd probably extract it to a constant
2. 30 seconds seems long. I can easily see someone responding "yes"/"ok"/"no" or similar to two separate messages in that time span. how about 10s?

Reply:
Might be worth tweaking for sure, but I'd rather we try it and live with it for a little while and only then change it

Reply:
Sorry but Im kinda agreeing with jarek here Régis, 30 seconds seems a little bit long. What are we protecting against here?

- a user spamming? 10 seconds would be enough
- some weird network error causing a duplicate? 10 seconds would be enough

In what situation is 30 seconds better than 10, considering the risk of 30 seconds Jarek exposed?

Reply:
Not a hill I want to die on. 10 seconds it is 👍